crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -87,9 +56,37 @@ impl MongoDb {
 #[async_trait]
 impl Database for MongoDb {
     #[tracing::instrument(skip(self), fields(function_type = "db_call"))]
-    async fn create_job(&self, job: JobItem) -> Result<JobItem> {
-        self.get_job_collection().insert_one(&job, None).await?;
-        Ok(job)
+    async fn create_job(&self, job: JobItem) -> Result<JobItem, JobError> {
+        let options = UpdateOptions::builder().upsert(true).build();
+
+        let updates = job.to_document().map_err(|e| JobError::Other(e.into()))?;
```

Comment:
IMO, can we not read the job_type and internal_id from job itself, is it necessary to get it from updates ?
I would prefer to do less operations on Document and more on typed Structs.
WDYT?

Reply:
I tried that first but I had to convert that job type to a bson as well to use inside `doc!`, so since the conversion was already happening above I can use it directly.